src/Compilers/CSharp/Test/Symbol/Symbols/GenericConstraintTests.cs

```diff
@@ -2100,6 +2100,9 @@ partial class E<T> where T : I, B { }
                 // (6,30): error CS0717: 'S': static classes cannot be used as constraints
                 // partial class C<T> where T : S { }
                 Diagnostic(ErrorCode.ERR_ConstraintIsStaticClass, "S").WithArguments("S").WithLocation(6, 30),
+                // (7,15): error CS0265: Partial declarations of 'D<T>' have inconsistent constraints for type parameter 'T'
```

Comment:
>error CS0265: Partial declarations of 'D<T>' have inconsistent constraints for type parameter 'T' [](start = 27, length = 97)

I didn't understand this error. The three partial declarations for `D<T>` seem to agree: `A, I`, `I, A` and `I, A`.

Reply:
> I didn't understand this error. The three partial declarations for D<T> seem to agree: A, I, I, A and I, A.

See the error below. Invalid constraint types are removed from the set before partials are matched, therefore other partials do not have 'A' among the constraints. I believe this behavior matches C# 7.3 compiler

---
In reply to: [280533172](https://github.com/dotnet/roslyn/pull/35375#discussion_r280533172) [](ancestors = 280533172)

Reply:
Thanks

---
In reply to: [280541572](https://github.com/dotnet/roslyn/pull/35375#discussion_r280541572) [](ancestors = 280541572,280533172)